test/router.ts

```diff
@@ -0,0 +1,317 @@
+var assert = require("assert");
+var router = require("../lib/router");
+
+describe("router.Router", () => {
+  describe("constructor", () => {
+    it("parses well formatted configs", () => {
+      process.env.SCHOOL = "Hogwarts";
+      const conf = `
+routes:
```

Comment:
Interesting thought.  We were actually talking about adding version numbers to the `schema_definitions.json` file.  Adding a version number to the kv_config does make sense  though.  I'll do that another commit.